src/session_impl.cpp

```diff
@@ -1763,6 +1763,55 @@ namespace {
 		reopen_listen_sockets();
 	}
 
+	void session_impl::interface_to_endpoints(std::string const& device, int const port
+		, bool const ssl, std::vector<listen_endpoint_t>& eps)
```

Comment:
do callers of this function rely on ``eps`` being appended to, and not replaced? if not, it seems it would be better to return the vector.
Really, unless you expect this to have a material performance impact, that interface probably would make sense regardless (and callers can make it obvious that they append the results)